lametro/utils.py

```diff
@@ -16,24 +17,17 @@ def format_full_text(full_text):
     results = ''
 
     if full_text:
-        txt_as_array = full_text.split("..")
-        for item in txt_as_array:
-            if 'SUBJECT:' in item:
-                array_with_subject = item.split('\n\n')
-                for item in array_with_subject:
-                    if 'SUBJECT:' in item:
-                        results = item.replace('\n', '')
+        full_text_with_cue = full_text.replace('\n\n', 'NEWLINE').replace('\r\n', 'NEWLINE').replace('\n..', 'NEWLINE')
+
+        match = re.search('(SUBJECT:)(.*?)(NEWLINE|ACTION:)', full_text_with_cue.replace('\n', ' '))
```

Comment:
Why do this replace here, instead of in the `full_text_with_cue` definition?

Reply:
Yes, it seems like something like this would be better:

```
clean_full_text = full_text.replace('\n\n', 'NEWLINE').replace('\r\n', 'NEWLINE').replace('\n..', 'NEWLINE').replace('\n', ' ')

match = re.search('(SUBJECT:)(.*?)(NEWLINE|ACTION:)', clean_full_text)
```

Honestly, I think there must be a cleverer use of RegEx to get the desired results. Have you had any luck with MULTILINE? https://docs.python.org/3.6/library/re.html#re.MULTILINE That presumably would eliminate the need for `replace`....but it did not behave as I expected.

Reply:
you know, i'm really terrible at regex and [tend to avoid them](http://regex.info/blog/2006-09-15/247).

what is the intent here? it seems like you are just looking for "SUBJECT" and either "NEWLINE" or "ACTION".

could you do `'SUBJECT' in clean_full_text and any(token in clean_full_text for token in ['NEWLINE', 'ACTION'])` or something like that?

Reply:
are new lines meaningful?

Reply:
Yes, I fear RegEx, too. [My previous solution](https://github.com/datamade/la-metro-councilmatic/blob/master/lametro/utils.py#L15) completely avoided RegEx, but it also missed some edge cases and was more difficult to decipher. 

I added some comments to give you a sense about how newlines help parse the subject header. @hancush - happy to chat about it!